### PR TITLE
feat: fatal error config not found

### LINKED ISSUE #10127: Fatal error if RENOVATE_CONFIG_FILE not found
**What would you like Renovate to be able to do?**

Fatal error & exit if RENOVATE_CONFIG_FILE is defined but the file is not found.

**Did you already have any implementation ideas?**

In here: https://github.com/renovatebot/renovate/blob/e0488a3ebbd8ddc5bbe3d1a303ec4cf31ad6cf7f/lib/config/file.ts#L7

### PR DESCRIPTION
**Pull request by weikangchia**
## Changes:

Fatal error & exit if RENOVATE_CONFIG_FILE is defined but the file is not found.

## Context:

Closes #10127

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/global/config/parse/file.spec.ts [hunk 1]
```diff
@@ -47,16 +48,20 @@ describe('workers/global/config/parse/file', () => {
...
+
       fs.unlinkSync(configFile);
     });
-  });
-  it('handles when invalid file location is provided', () => {
-    const configFile = upath.resolve(tmp.path, './file4.js');
-    expect(file.getConfig({ RENOVATE_CONFIG_FILE: configFile })).toStrictEqual(
-      {}
-    );
+    it('fatal error and exit if custom config file does not exist', () => {
```
Comment (viceice) [inline lib/workers/global/config/parse/file.spec.ts:56]:
```suggestion

    it('fatal error and exit if custom config file does not exist', () => {
```

File: lib/workers/global/config/parse/file.spec.ts [hunk 2]
```diff
@@ -33,7 +28,13 @@ describe('workers/global/config/parse/file', () => {
       expect(res).toMatchSnapshot();
       expect(res.rangeStrategy).toEqual('bump');
     });
-    it('informs user when error in parsing config.js', () => {
+    it('parse and returns empty config if there is no RENOVATE_CONFIG_FILE in env', () => {
```
Comment (viceice) [inline lib/workers/global/config/parse/file.spec.ts:31]:
```suggestion

    it('parse and returns empty config if there is no RENOVATE_CONFIG_FILE in env', () => {
```

Readabillity

File: lib/workers/global/config/parse/file.spec.ts [hunk 3]
```diff
@@ -33,7 +28,13 @@ describe('workers/global/config/parse/file', () => {
...
+      expect(file.getConfig({})).toEqual({});
+    });
+    it('fatal error and exit if error in parsing config.js', () => {
```
Comment (viceice) [inline lib/workers/global/config/parse/file.spec.ts:34]:
```suggestion

    it('fatal error and exit if error in parsing config.js', () => {
```

File: lib/workers/global/config/parse/file.ts [hunk 4]
```diff
@@ -18,9 +18,13 @@ export function getConfig(env: NodeJS.ProcessEnv): AllConfig {
     if (err instanceof SyntaxError || err instanceof TypeError) {
       logger.fatal(`Could not parse config file \n ${err.stack}`);
       process.exit(1);
+    } else if (env.RENOVATE_CONFIG_FILE) {
+      logger.fatal('No custom config file found on disk');
+      process.exit(1);
```
Comment (viceice) [inline lib/workers/global/config/parse/file.ts:23]: @rarkins Shouldn't we throw a config error instead? 🤔

Comment (rarkins) [reply lib/workers/global/config/parse/file.ts:23]: I don't mind. Exiting immediately is consistent with the earlier if too

File: lib/workers/global/config/parse/file.spec.ts (hunk 2 above)
Comment (weikangchia) [reply lib/workers/global/config/parse/file.spec.ts:32]: Updated code

File: lib/workers/global/config/parse/file.ts (hunk 4 above)
Comment (weikangchia) [reply lib/workers/global/config/parse/file.ts:23]: Do let me know your preference and I can help to make the change. Thanks!

Comment (rarkins) [reply lib/workers/global/config/parse/file.ts:23]: Let's keep it as is

**Review by viceice — APPROVED** [review]

**Pull request merged**